dimagi/commcare-hq
# Updated logging configuration to fix celery logging issues

## Product Description
Logging messages for generated reports were not showing up. This code aims to fix that.

## Technical Summary
https://dimagi-dev.atlassian.net/browse/SAAS-13755.  Although the original ticket was about generated reports, this seems like it would have affected all logs that were generated within celery tasks.  This seems to have occurred because [celery resets logging handlers on init](https://stackoverflow.com/a/48370737). The working fix is that we catch that celery log setup signal and just have celery use the ones all the rest of the code are using. (The celery settings flag to do this didn't seem to work).

Note that this fix also no longer disables existing loggers (this is the django-recommended setting, and with it set to True, no messages were showing up), and the celery task logger no longer propagates its messages (also shouldn't matter, as I'd assume that logger wasn't logging anything correctly, before).

## Feature Flag


## Safety Assurance

### Safety story
Tested locally and on staging.  An important thing to note, if testing on staging, is that celery will only reset the logging if it is being run as its own process, i.e. if `CELERY_TASK_ALWAYS_EAGER` is `False`.

### Automated test coverage
No tests

### QA Plan

I'm not sure how to handle QA here. I'm changing some of the general logging configuration (not disabling existing loggers), and so would like to ensure this doesn't break our existing setup. However, I don't think QA is going to catch that, as we should only see these results in the logs.

### Rollback instructions




### Labels & Review
